runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -112,14 +120,26 @@ public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
                 .recordStats()
                 .build();
 
-        MetricRegistry metrics = CorfuRuntime.getDefaultMetrics();
-        final String pfx = String.format("%s0x%x.cache.", CorfuComponent.ADDRESS_SPACE_VIEW.toString(),
-                this.hashCode());
-        metrics.register(pfx + "cache-size", (Gauge<Long>) readCache::size);
-        metrics.register(pfx + "evictions", (Gauge<Long>) () -> readCache.stats().evictionCount());
-        metrics.register(pfx + "hit-rate", (Gauge<Double>) () -> readCache.stats().hitRate());
-        metrics.register(pfx + "hits", (Gauge<Long>) () -> readCache.stats().hitCount());
-        metrics.register(pfx + "misses", (Gauge<Long>) () -> readCache.stats().missCount());
+        Optional<MeterRegistry> metricsRegistry = runtime.getRegistry();
+
+        metricsRegistry.ifPresent(registry ->
+                GuavaCacheMetrics.monitor(registry, readCache, "address_space.read_cache"));
```

Comment:
Can you add a gauge on the miss ratio and some metrics for the load time stats.

Reply:
done